packages/trader/src/Stores/Modules/Trading/Actions/__tests__/start-date.spec.ts

```diff
@@ -0,0 +1,276 @@
+        barriers: 1,
+        contract_category: 'vanilla',
+        contract_category_display: 'Vanilla Options',
+        exchange_name: 'RANDOM',
+        market: 'synthetic_index',
+        start_type: 'spot',
+        submarket: 'random_index',
+        underlying_symbol: 'R_10',
+    };
+
+    return {
+        ...jest.requireActual('@deriv/shared'),
+        WS: {
+            storage: {
+                contractsFor: jest.fn(() =>
+                    Promise.resolve({
+                        contracts_for: {
+                            available: [
+                                {
+                                    ...vanilla_contracts_for_data,
+                                    barrier: barrier_daily,
+                                    barrier_choices: barrier_choices_daily,
+                                    contract_display: 'Vanilla Long Call',
+                                    contract_type: 'VANILLALONGCALL',
+                                    expiry_type: 'daily',
+                                    max_contract_duration: '365d',
+                                    min_contract_duration: '1d',
+                                    sentiment: 'up',
+                                },
+                                {
+                                    ...vanilla_contracts_for_data,
+                                    barrier: barrier_daily,
+                                    barrier_choices: barrier_choices_daily,
+                                    contract_display: 'Vanilla Long Put',
+                                    contract_type: 'VANILLALONGPUT',
+                                    expiry_type: 'daily',
+                                    max_contract_duration: '365d',
+                                    min_contract_duration: '1d',
+                                    sentiment: 'down',
+                                },
+                                {
+                                    ...vanilla_contracts_for_data,
+                                    barrier: barrier_intraday,
+                                    barrier_choices: barrier_choices_intraday,
+                                    contract_display: 'Vanilla Long Call',
+                                    contract_type: 'VANILLALONGCALL',
+                                    expiry_type: 'intraday',
+                                    max_contract_duration: '1d',
+                                    min_contract_duration: '1m',
+                                    sentiment: 'up',
+                                },
+                                {
+                                    ...vanilla_contracts_for_data,
+                                    barrier: barrier_intraday,
+                                    barrier_choices: barrier_choices_intraday,
+                                    contract_display: 'Vanilla Long Put',
+                                    contract_type: 'VANILLALONGPUT',
```

Comment:
Probably, same in this file

Reply:
same as https://github.com/binary-com/deriv-app/pull/12459#discussion_r1436302828